lib/models/__tests__/organization.test.ts

```diff
@@ -0,0 +1,202 @@
+// tslint:disable:no-any
+
+import * as DocumentDb from "documentdb";
+
+import * as DocumentDbUtils from "../../utils/documentdb";
+import {ModelId} from "../../utils/documentdb_model_versioned";
+import { toNonNegativeNumber } from "../../utils/numbers";
+
+import {IOrganization, IRetrievedOrganization, OrganizationModel} from "../organization";
+
+const aDatabaseUri = DocumentDbUtils.getDatabaseUri("mockdb");
+const organizationsCollectionUrl = DocumentDbUtils.getCollectionUri(aDatabaseUri, "organizations");
+
+const organizationId = "xyz" as ModelId;
+
+const aRetrievedOrganization: IRetrievedOrganization = {
+    _self: "xyz",
+    _ts: "xyz",
+    organizationId: organizationId,
+    name: "MyOrganization",
+    id: "xyz",
+    kind: "IRetrievedOrganization",
+    version: toNonNegativeNumber(0).get,
+};
+
+describe("findOneOrganizationById", () => {
+
+    it("should resolve a promise to an existing organization", async () => {
+        const iteratorMock = {
+            executeNext: jest.fn((cb) => cb(undefined, ["result"], undefined)),
+        };
+
+        const clientMock = {
+            queryDocuments: jest.fn((__, ___) => iteratorMock),
+        };
+
+        const model = new OrganizationModel((clientMock as any) as DocumentDb.DocumentClient, organizationsCollectionUrl);
+
+        const result = await model.findByOrganizationId("id");
+
+        expect(result.isRight).toBeTruthy();
+        if (result.isRight) {
+            expect(result.right.isDefined).toBeTruthy();
+            expect(result.right.get).toEqual("result");
+        }
+    });
+
+    it("should resolve a promise to undefined if no organization is found", async () => {
```

Comment:
Actually we need to change this description (also in the other tests) since I've opted for `Options` instead of `undefined` values.

It should say `should resolve a promise to an empty value if no organization is found`